src/Test/Utilities/Desktop/TestHelpers.cs

```diff
@@ -123,5 +125,25 @@ public class TestAnalyzer : DiagnosticAnalyzer
 
             return dir.CreateFile(assemblyName + ".dll").WriteAllBytes(analyzerCompilation.EmitToArray());
         }
+
+        /// <summary>
+        /// <see cref="System.Xml.Linq.XComment.Value"/> is serialized with "--" replaced by "- -"
+        /// </summary>
+        public static string AsXmlCommentText(string text)
+        {
+            var builder = new StringBuilder();
```

Comment:
Using `string.Replace` would probably require multiple calls to handle `"---"`. The current implementation is more verbose but maybe clearer. Will leave as is.